src/providers/structure.ts

```diff
@@ -68,6 +56,419 @@ export class SectionNodeProvider implements vscode.TreeDataProvider<Section> {
+            }
+            return
+        })
+        if (!ast) {
+            return []
+        }
+
+        // Parse each base-level node. If the node has contents, that function
+        // will be called recursively.
+        let sections: Section[] = []
+        for (const node of ast.content) {
+            sections = [
+                ...sections,
+                ...await this.parseLaTeXNode(node, file, subFile, filesBuilt, commands, depths)
+            ]
+        }
+
+        return sections
+    }
+
+    /**
+     * This function parses a particular LaTeX AST node and its sub-nodes
+     * (contents by `latex-utensils`).
+     *
+     * @param node The AST node to be parsed.
+     *
+     * All other parameters are identical to {@link buildLaTeXSectionFromFile}.
+     *
+     * @returns A flat array of {@link Section} of this node.
+     */
+    private async parseLaTeXNode(node: latexParser.Node, file: string, subFile: boolean, filesBuilt: Set<string>, commands: {cmds: string[], envs: string[], secs: string[]}, depths: {[cmd: string]: number}): Promise<Section[]> {
+        let sections: Section[] = []
+        if (latexParser.isCommand(node)) {
+            if (commands.secs.includes(node.name.replace(/\*$/, ''))) {
+                // \section{Title}
+                sections.push(new Section(
+                    node.name.endsWith('*') ? SectionKind.NoNumberSection : SectionKind.Section,
+                    this.captionify(node.args[node.args.length - 1]),
+                    vscode.TreeItemCollapsibleState.Expanded,
+                    depths[node.name.replace(/\*$/, '')],
+                    node.location.start.line - 1,
+                    node.location.end.line - 1,
+                    file
+                ))
+            } else if (commands.cmds.includes(node.name.replace(/\*$/, ''))) {
+                // \notlabel{Show}{ShowAlso}
+                const caption = node.args.map(arg => {
+                    const argContent = latexParser.stringify(arg)
+                    return argContent.slice(1, argContent.length - 1)
+                }).join(', ') // -> Show, ShowAlso
+                sections.push(new Section(
+                    SectionKind.Label,
+                    `#${node.name}: ${caption}`,
+                    vscode.TreeItemCollapsibleState.Expanded,
+                    -1,
+                    node.location.start.line - 1,
+                    node.location.end.line - 1,
+                    file
+                ))
+            } else if (subFile) {
+                // Check if this command is a subfile one
+                sections = [
+                    ...sections,
+                    ...await this.parseLaTeXSubFileCommand(node, file, subFile, filesBuilt, commands, depths)
+                ]
+            }
+        } else if (latexParser.isLabelCommand(node) && node.name === 'label') {
```

Comment:
In the previous version, not showing labels was possible by simply removing `label` from `latex-workshop.view.outline.commands`. It would be nice to follow the same lines here.